### ISSUE TITLE
Default ApplicationInsights.config filters out most telemetry when using OWIN

### ISSUE THREAD

**Issue by gregjhogan** [Microsoft/ApplicationInsights-server-dotnet#16]
When using OWIN with the default ApplicationInsights.config all telemetry sent from inside the OWIN pipeline seems to be filtered out (and never makes it up to AI).  I am guessing it is because of the below default configuration when adding application insights to a project:

```xml
<TelemetryProcessors>
  ...
  <Add Type="Microsoft.ApplicationInsights.Web.HandlerTelemetryProcessor, Microsoft.AI.Web">
    <Handlers>
      ...
      <Add Value="System.Web.Handlers.TransferRequestHandler"/>
      ...
    </Handlers>
  </Add>
</TelemetryProcessors>
```

For example, the out-of-the-box request tracking (RequestTelemetry) gets filtered out, and any custom telemetry sent, too.

**Comment by SergeyKanzhelev** [comment]
Is it regular OWIN app? We have a functional test that is passing

Test: https://github.com/Microsoft/ApplicationInsights-server-dotnet/blob/aac44ef4cf526143d71683464e1c5a956cdeb5df/Test/Web/FunctionalTests/FunctionalTests/WebAppFW45SampledTests.cs#L63


Test app: https://github.com/Microsoft/ApplicationInsights-server-dotnet/blob/aac44ef4cf526143d71683464e1c5a956cdeb5df/Test/Web/FunctionalTests/TestApps/WebAppFW45/Controllers/ProductsController.cs

Can you give us more details on what FW version do you use?

**Comment by gregjhogan** [comment]
The application I am seeing the issue in is much more complex than this, but here is an easy way to reproduce the issue ... create a new web project with the following nuget packages and startup file:

packages.config
```xml
<?xml version="1.0" encoding="utf-8"?>
<packages>
  <package id="Microsoft.ApplicationInsights" version="2.0.0-beta4" targetFramework="net461" />
  <package id="Microsoft.ApplicationInsights.Agent.Intercept" version="1.2.1" targetFramework="net461" />
  <package id="Microsoft.ApplicationInsights.DependencyCollector" version="2.0.0-beta4" targetFramework="net461" />
  <package id="Microsoft.ApplicationInsights.PerfCounterCollector" version="2.0.0-beta4" targetFramework="net461" />
  <package id="Microsoft.ApplicationInsights.Web" version="2.0.0-beta4" targetFramework="net461" />
  <package id="Microsoft.ApplicationInsights.WindowsServer" version="2.0.0-beta4" targetFramework="net461" />
  <package id="Microsoft.ApplicationInsights.WindowsServer.TelemetryChannel" version="2.0.0-beta4" targetFramework="net461" />
  <package id="Microsoft.CodeDom.Providers.DotNetCompilerPlatform" version="1.0.1" targetFramework="net461" />
  <package id="Microsoft.Net.Compilers" version="1.1.1" targetFramework="net461" developmentDependency="true" />
  <package id="Microsoft.Owin" version="3.0.1" targetFramework="net461" />
  <package id="Microsoft.Owin.Host.SystemWeb" version="3.0.1" targetFramework="net461" />
  <package id="Owin" version="1.0" targetFramework="net461" />
</packages>
```

Startup.cs
```cs
namespace WebApplication2
{
    using System.Web;
    using Microsoft.ApplicationInsights;
    using Owin;

    public class Startup
    {
        public void Configuration(IAppBuilder app)
        {
            app.Run(context =>
            {
                new TelemetryClient().TrackEvent("IHttpHandler " + HttpContext.Current.Handler);

                context.Response.ContentType = "text/plain";
                return context.Response.WriteAsync("Hello World! ");
            });
        }
    }
}
```

The above telemetry call doesn't get sent to app insights.  If I comment out the app insights XML config line where it adds the TransferRequestHandler the message gets sent out successfully.

I found this really confusing because in visual studio diagnostic tools you see the app insights event, and it looks like everything is good.  Then, I was left scratching my head trying to figure out why telemetry isn't showing up in the portal (especially when the AI status blog says there are delays processing telemetry).  Maybe I am missing something; what is the purpose of filtering out this handler by default?

**Comment by swathisundar** [comment]
I am seeing this problem as well. 

In our packages.config, we have the following
          <package id="Microsoft.ApplicationInsights.Web" version="1.2.3" targetFramework="net45" />
And we have added 
       <Add Type="Microsoft.ApplicationInsights.Web.RequestTrackingTelemetryModule, Microsoft.AI.Web"/>.

However, we do not see any requests logged in the Azure portal. We do see logs for dependencies though. 

Referring to http://apmtips.com/blog/2015/02/23/request-name-and-url/, “There is no smart request name calculation for attributes-based routing today”.

We tried using the beta, pre-release version
  <package id="Microsoft.ApplicationInsights.Web" version="2.0.0-beta4" targetFramework="net45" />
And with this, **I see that the requests are being pushed to AI in the VS output window, however there are not logged in the portal.**
